### DISCUSSION TITLE
How to save timedeltas in database

### DISCUSSION THREAD

**Discussion by thisiseddy-ab** [sqlalchemy/sqlalchemy#8279] [Usage Questions] [answered]
I need to save video duration in database, but i want to save it as timedelta so i can to "<>" operation, is it possible to save timedelta in database should i use DateTime type ?

I Found a Solution to convert it to an ISO duration and saving as a string.

```py
import isodate
td = timedelta(hours=1,minutes=45,seconds=25)
### Converting to ISO Duration
duration_isoformat = isodate.duration_isoformat(td)
print(duration_isoformat)
### Converting back to Timedelta
duration_temedelta = isodate.parse_duration(duration_isoformat)
print(duration_temedelta)
```

Is there a better way ?

**Comment by CaselIT** [comment, marked as answer]
Hi,

An alternative to the option you provided could also be to save the as numeric number of seconds (optionally with fractional second too) using int/float/numeric

Some db also have specific types for timedelta, but it's not specified what the db will be

**Reply by thisiseddy-ab** [reply]
im using postrgresdb

**Reply by CaselIT** [reply]
Then postgresql also supports the interval type: https://www.postgresql.org/docs/current/datatype-datetime.html
